### PR TITLE
Create new strings job with unicode/ascii tasks

### PR DESCRIPTION
**Pull request by aarontp**
Note, this is the first Job that actually has more than one Task.   Also adds a TextFile Evidence type.

### CODE REVIEW

File: turbinia/evidence.py [hunk 1]
```diff
@@ -290,3 +290,11 @@ def __init__(self, text_data=None, *args, **kwargs):
...
     super(ReportText, self).__init__(*args, **kwargs)
     self.copyable = True
+
+
+class TextFile(Evidence):
+  """Text data."""
+
+  def __init__(self, *args, **kwargs):
+    super(TextFile, self).__init__(*args, **kwargs)
+    self.copyable = True
```
Comment (berggren) [inline turbinia/evidence.py]: I don't remember what "copyable" means?

File: turbinia/jobs/strings.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+from turbinia.evidence import GoogleCloudDiskRawEmbedded
+from turbinia.evidence import RawDisk
+from turbinia.evidence import TextFile
+from turbinia.jobs import TurbiniaJob
+from turbinia.workers.strings import StringsAsciiTask
+from turbinia.workers.strings import StringsUniTask
+
+
+class StringsJob(TurbiniaJob):
+
```
Comment (berggren) [inline turbinia/jobs/strings.py]: Add docstring

File: turbinia/workers/strings.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    # Add a log line to the result that will be returned.
+    result.log('Running strings as [{0:s}]'.format(cmd))
+    # Actually execute the binary
+    self.execute(
+        cmd, result, new_evidence=[output_evidence], close=True, shell=True)
+
+    return result
+
+
+class StringsUniTask(TurbiniaTask):
```
Comment (berggren) [inline turbinia/workers/strings.py]: No need to shorten the name, I would prefer "StringsUnicodeTask"

File: turbinia/workers/strings.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+    Args:
+        evidence (Evidence object):  The evidence we will process
+        result (TurbiniaTaskResult): The object to place task results into.
+
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    # Create the new Evidence object that will be generated by this Task.
+    output_evidence = TextFile()
```
Comment (berggren) [inline turbinia/workers/strings.py]: Do we want to output this generic evidence type or create something like "StringsUnicodeTextFile" so other tasks can pick it up and do targeted analysis?

File: turbinia/evidence.py (hunk 1 above)
Comment (aarontp) [reply turbinia/evidence.py]: You have seen it by now, but it's documented in https://github.com/google/turbinia/pull/209

File: turbinia/jobs/strings.py (hunk 2 above)
Comment (aarontp) [reply turbinia/jobs/strings.py]: Done.

File: turbinia/workers/strings.py (hunk 3 above)
Comment (aarontp) [reply turbinia/workers/strings.py]: Sure.  Done.

File: turbinia/workers/strings.py (hunk 4 above)
Comment (aarontp) [reply turbinia/workers/strings.py]: As discussed offline, I think we'll keep it as is.  We can use the evidence.processed_by (list) to check what the parent jobs are if/when needed instead.

**Pull request merged**